flexmeasures/data/models/reporting/cost.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any, List, Dict
+
+
+from flexmeasures.data.models.reporting import Reporter
+from flexmeasures.data.schemas.reporting.cost import (
+    CostReporterConfigSchema,
+    CostReporterParametersSchema,
+)
+from flexmeasures.data.models.time_series import Sensor
+from flexmeasures.utils.time_utils import server_now
+from flexmeasures.data.queries.utils import simplify_index
+
+
+class CostReporter(Reporter):
+    """Compute the cashflow due to energy/power flow.
+
+    Given power/energy and price sensors, this reporter computes the cost/profit
+    of a power/energy flow under a certain tariff.
+
+    Sign convention
+    ----------------
+
+    Power flows:
+        (+) production
+        (-) consumption
+
+    Cash flows:
```

Comment:
True! This made me consider renaming the reporter to  `ProfitReporter`. This is because I see to have negative cost/revenue kind of odd. I would expect those reporters to report non-negative quantities.